object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -490,16 +504,24 @@ private void handleAssignFromRHS(
         LocalVariableNode mustCallChoiceParamLocal = getLocalPassedAsMustCallChoiceParam(rhs);
         if (mustCallChoiceParamLocal != null) {
           if (isVarInDefs(newDefs, mustCallChoiceParamLocal)) {
-            ImmutableSet<LocalVarWithTree> immutableSet =
+            LocalVarWithTree latestAssignmentPair =
+                getAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
+            // If both side of this assignment node point to the same resource (rhs is MCC with the
+            // lhs), then we will replace the previous assignment tree of lhs with current
+            // assignment tree, otherwise we will just add a new LocalVarWithTree to the set that
+            // contains mustCallChoiceParamLocal
+            ImmutableSet<LocalVarWithTree> setContainingMustCallChoiceParamLocal =
                 getSetContainingAssignmentTreeOfVar(newDefs, mustCallChoiceParamLocal);
-            ImmutableSet<LocalVarWithTree> newImmutableSet =
-                FluentIterable.from(immutableSet)
+            ImmutableSet<LocalVarWithTree> newSetContainingMustCallChoiceParamLocal =
+                FluentIterable.from(setContainingMustCallChoiceParamLocal)
+                    .filter(Predicates.not(Predicates.equalTo(latestAssignmentPair)))
```

Comment:
Is the idea here that if `latestAssignmentPair` is `null`, nothing will get filtered by the `filter` call, which is the right thing to do?  If so, we should document that.  Right now the code is a bit too clever 🙂

Reply:
@Nargeshdb any thoughts on this one?

Reply:
> Is the idea here that if `latestAssignmentPair` is `null`, nothing will get filtered by the `filter` call, which is the right thing to do?
>
Yes, exactly.
> If so, we should document that. Right now the code is a bit too clever 🙂
>
I added a comment here.

Reply:
Sorry I meant a comment specifically about the null case

Reply:
https://github.com/kelloggm/object-construction-checker/pull/282/commits/b3b069b47173c8ec3a3cb8e34cd988d7be04a85a